overlord/devicestate/devicestate.go

```diff
@@ -385,6 +407,26 @@ func Remodel(st *state.State, new *asserts.Model) ([]*state.TaskSet, error) {
 			return nil, err
 		}
 	}
+	// unmark no-longer required snaps
+	requiredSnaps := getAllRequiredSnapsForModel(new)
+	snapStates, err := snapstate.All(st)
+	for snapName, snapst := range snapStates {
+		// TODO: remove this type restriction once we remodel
+		//       bases/kernels/gadgets and add tests that ensure
+		//       that the required flag is properly set/unset
+		typ, err := snapst.Type()
```

Comment:
What is the origin of the type restriction? The comment doesn't make this obvious.

Reply:
we cannot do it here, it would not be undone,  we need to do it in the task where we switch to the new model assertion